### ISSUE TITLE
[3.0] Setting listenOnlyMode to false has no effect

### ISSUE THREAD

**Issue by Ithanil** [bigbluebutton/bigbluebutton#21934]
**Describe the bug**
Setting `public.app.listenOnlyMode` to false seemingly has no effect, no matter if the setting is overriden via `/etc/bigbluebutton/bbb-html5.yml` or if it is set directly in `/usr/share/bigbluebutton/html5-client/private/config/settings.yml`.

**Expected behavior**
With `listenOnlyMode: false`, I would expect to directly get to the audio configuration modal on join.

**Actual behavior**
I still get the choice between both audio modes.

**BBB version:**
3.0 Beta 7

**Desktop (please complete the following information):**
Linux Firefox / Chromium

**Comment by Ithanil** [comment]
I've tried on an older (beta 5) server, without LiveKit, and get the same result, after just changing `true` to `false` in `/usr/share/bigbluebutton/html5-client/private/config/settings.yml` and then a `bbb-conf --restart`. Another `systemctl restart bbb-apps-akka` does not help.

Passing `userdata-bbb_listen_only_mode=false` results in the expected behavior. What could be going on here?

**Comment by Ithanil** [comment]
@schrd also can't reproduce this, so apparently there is something weird on my end. Still have no idea what.

**Comment by Ithanil** [comment]
@prlanzarin Sooo, I've tried * a lot * of things, but really nothing in terms of configuration had any effect. In the end it turned out the only thing that made the setting `listenOnlyMode: false` fully effective was a full **reboot** of the system. This goes both ways, i.e. for disabling and enabling the listenOnly option. However, a simple `bbb-conf --restart` is enough to achieve one part of the effect, namely preventing to actually get into listen-only audio (selecting on listen-only will join fullaudio). To also prevent the selection modal from popping up, I needed the reboot (a.k.a. restart the proper services).

So, in conclusion I think there is something missing from `bbb-conf --restart` in BBB 3.0 (BBB 2.7 worked as expected).

Does this make any sense to you?

**Comment by imdt-claudiop** [comment]
On BigBlueButton 4.0 (`v4.0.x-develop`, commit `93bc4864`) this cannot reproduce as described, and
the reason is worth recording because it inverts the report.

In `bigbluebutton-html5/imports/ui/components/audio/audio-modal/container.jsx`:

```js
const listenOnlyMode = forceListenOnly
  || (getFromUserSettings('bbb_listen_only_mode', APP_CONFIG.listenOnlyMode) && !usingLiveKit);
```

`usingLiveKit` is `meeting.audioBridge === 'livekit'`, and `audioBridge=livekit` is the default in
`bigbluebutton.properties` on 4.0. So on a default install the setting is short-circuited to false
before it is ever consulted, and the listen-only option is simply never offered in the audio modal.

Measured both ways on a stock 4.0 server, and the outcome is identical:

- shipped default (`public.app.listenOnlyMode: true`): no listen-only option in the modal
- `userdata-bbb_listen_only_mode=false` on the join URL: no listen-only option in the modal

So the reported symptom (set it to false, still get the choice) does not occur on 4.0, but the
underlying complaint that the setting is ignored is true at this commit and true in both directions:
setting it to `true` has no effect either. The `&& !usingLiveKit` condition dates from
`3c2339fece` (2025-03-26) and predates 4.0.

That also fits what Ithanil observed at the time. On his LiveKit-enabled server the userdata param
behaved and the config did not, which is what you would expect if the config path was being
overridden rather than not read.

If the intent is that this flag should still mean something on the LiveKit bridge, that is a separate
change from what this issue describes. If the intent is that listen-only is simply gone on LiveKit,
then the flag is dead configuration on a default install and this issue is closable for 4.0.
